ca/certificates.go

```diff
@@ -209,23 +200,60 @@ func (rca *RootCA) RequestAndSaveNewCertificates(ctx context.Context, paths Cert
 		return nil, err
 	}
 
-	// Ensure directory exists
-	err = os.MkdirAll(filepath.Dir(paths.Cert), 0755)
+	kekUpdate, err := rca.getKEKUpdate(ctx, X509Cert, tlsKeyPair, r)
 	if err != nil {
 		return nil, err
 	}
 
-	// Write the chain to disk
-	if err := ioutils.AtomicWriteFile(paths.Cert, signedCert, 0644); err != nil {
+	if err := kw.Write(signedCert, key, kekUpdate); err != nil {
 		return nil, err
 	}
 
-	// Move the new key to the final location
-	if err := os.Rename(tempPaths.Key, paths.Key); err != nil {
+	return &tlsKeyPair, nil
+}
+
+func (rca *RootCA) getKEKUpdate(ctx context.Context, cert *x509.Certificate, keypair tls.Certificate, r remotes.Remotes) (*KEKUpdate, error) {
+	var managerRole bool
+	for _, ou := range cert.Subject.OrganizationalUnit {
+		if ou == WorkerRole {
+			// If this is a worker set to never encrypt. We always want to set to nil, in case this was a manager that was demoted
+			// to a worker.
```

Comment:
Er, sorry, the KEK.  We want KEK to always be empty for a worker, because we don't want the tls key encrypted unless it's a manager (at least that was my understanding from our previous discussion - I'm happy to add worker TLS encryption as well, but we'd have to add some way to stream worker KEK changes down to the agent).